deep-learning/_posts/2017-08-28-compositional-spatial-transformer.md

```diff
@@ -7,28 +7,30 @@ cite:
     authors: "Chen-Hsuan Lin and Simon Lucey"
     title:  "Inverse Compositional Spatial Transformer Networks" 
     venue:   "CVPR 2017"
-pdf: "Inverse Compositional Spatial Transformer Networks"
+pdf: "https://arxiv.org/pdf/1612.03897.pdf"
 ---
    
 ## Summary
 
-The authors present a method to improve the robustness of a CNN to spatial variations. Instead of using a transformation nerwork [1] as in Fig.1 which tries to recover the proper warping function $$p$$ given the input image I, they implemented an iterative approach inspired by Lucas-Kanade optical flow algorithm. 
+The authors present a method to improve the robustness of a CNN to spatial variations. Instead of using a transformation nerwork [1] as in Fig.1 which tries to recover the warping function $$p$$ given the input image $$I_{in}$$, they implemented an iterative approach inspired by Lucas-Kanade optical flow algorithm. 
 
 ![](/deep-learning/images/icstn/sc.png)
 
 The Lucas-Kanade algorithm tries to minimize the sum of squared differences
-(SSD) objective $$ min_{\delta_p} ||I(p+\delta_p) - T(0)|| $$ where $$I$$ is the input image, $$p $$ is the transformation and $$T(0)$$ is a template image.  Since the Lucas-kanade algorithm is iterative ($$p=p+\delta_p$$), the proposed method implements a sequence of transformations networks.
+(SSD) objective $$ min_{\delta_p} ||I_{in}(p+\delta_p) - T(0)|| $$ where $$I_{in}$$ is the input image, $$p $$ is the transformation and $$T(0)$$ is a template image.  Since the Lucas-kanade algorithm is iterative ($$p=p+\delta_p$$), the proposed method implements a sequence of transformations networks.  So instead of trying to compute $$p$$ right away as for the STN, it computes a sequence of $$\delta_p$$.
 
 ![](/deep-learning/images/icstn/sc1.png)
  
-which can be represented as a recurrent network
+
+The network in Fig.4 can be represented as a recurrent network
 
 ![](/deep-learning/images/icstn/sc2.png)
 
 
 ## Experiments and results
 
 The authors show that the use of a recurrent transformation network gives better results while preventing from the boundary effect.
+
 ![](/deep-learning/images/icstn/sc3.png)
 
 ![](/deep-learning/images/icstn/sc4.png)
```

Comment:
Found code available at https://github.com/ericlin79119/IC-STN. It uses Python/Tensorflow :)